src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -4354,16 +4379,93 @@ private MemberResolutionResult<TMember> IsApplicable<TMember>(
                 isMethodGroupConversion: isMethodGroupConversion,
                 useSiteInfo: ref useSiteInfo);
             return new MemberResolutionResult<TMember>(member, leastOverriddenMember, applicableResult, hasTypeArgumentsInferredFromFunctionType);
+
+            static TMember construct(TMember member, ImmutableArray<TypeWithAnnotations> typeArguments)
+            {
+                if (member is MethodSymbol method)
+                {
+                    if (method.GetIsNewExtensionMember())
+                    {
+                        NamedTypeSymbol extension = method.ContainingType;
+                        if (extension.Arity > 0)
+                        {
+                            extension = extension.Construct(typeArguments[..extension.Arity]);
+                            method = method.AsMember(extension);
+                        }
+
+                        if (method.Arity > 0)
+                        {
+                            return (TMember)(Symbol)method.Construct(typeArguments[extension.Arity..]);
+                        }
+
+                        return (TMember)(Symbol)method;
+                    }
+
+                    return (TMember)(Symbol)method.Construct(typeArguments);
+                }
+
+                if (member is PropertySymbol property)
+                {
+                    Debug.Assert(property.GetIsNewExtensionMember());
+                    NamedTypeSymbol extension = property.ContainingType;
+                    Debug.Assert(extension.Arity > 0);
+
+                    extension = extension.Construct(typeArguments[..extension.Arity]);
```

Comment:
>typeArguments[..extension.Arity] [](http://example.com/codeflow?start=52&length=32)

```typeArguments```? We want to use all type arguments and would like to be aware when that is not the case.